### PR TITLE
Random mob consensus

### PR DESCRIPTION
**Pull request by knoellle**
## Introduced Changes

- Change scoring model for ransac samples to weight based on distance instead of simply counting points within some margin.
- Allow using separate margins for score and removing the points from the pool after a line was found.

Fixes #

This PR is part of a set:
- #683
- #684
- #686
- #687

 ~~Collect them all for a chance to win a golden PR! Buy now and win!~~
If you wish to test them all together, use this branch: [fliewatüüt](https://github.com/knoellle/hulk/tree/fliewat%C3%BC%C3%BCt)

## Ideas for Next Iterations (Not This PR)

Get rid of ransac.

## How to Test

Line detection should be more stable, especially when there are many points spread across the width of a line, it should find the center of the line much better now.
Furthermore, ransac should now be discouraged from cutting corners by the changed scoring model.

### CODE REVIEW

File: crates/vision/src/field_border_detection.rs [hunk 1]
```diff
@@ -112,13 +112,21 @@ fn find_border_lines(
     second_line_association_distance: f32,
 ) -> Vec<Line2> {
     // first line
-    let result = ransac.next_line(20, first_line_association_distance);
+    let result = ransac.next_line(
+        20,
```
Comment (okiwi6) [inline crates/vision/src/field_border_detection.rs:116]: Not your changes, but maybe we can make this a parameter? :)

**Review by okiwi6 — COMMENTED** [review]
Will continue review tomorrow

File: crates/vision/src/line_detection.rs [hunk 2]
```diff
@@ -38,8 +38,10 @@ pub struct CycleContext {
     gradient_alignment: Parameter<f32, "line_detection.$cycler_instance.gradient_alignment">,
     maximum_distance_to_robot:
         Parameter<f32, "line_detection.$cycler_instance.maximum_distance_to_robot">,
-    maximum_fit_distance_in_pixels:
+    maximum_fit_distance_in_field:
```
Comment (okiwi6) [inline crates/vision/src/line_detection.rs:41]: rebasing error?

File: crates/vision/src/ransac.rs [hunk 3]
```diff
@@ -24,34 +24,42 @@ impl Ransac {
...
+                    .filter(|&point| {
+                        line.squared_distance_to_point(*point) <= maximum_score_distance_squared
                     })
-                    .count();
+                    .map(|point| 1.0 - line.distance_to_point(*point) / maximum_score_distance)
+                    .sum();
                 (line, score)
             })
-            .max_by_key(|scored_line| scored_line.1)
+            .max_by(|line_1, line_2| line_1.1.total_cmp(&line_2.1))
```
Comment (okiwi6) [inline crates/vision/src/ransac.rs:58]: since score is not a member of a line, maybe prefer pattern matching here (`(_, score_1), ...`)

**Review by okiwi6 — CHANGES_REQUESTED** [review]

File: crates/vision/src/field_border_detection.rs (hunk 1 above)
Comment (knoellle) [reply crates/vision/src/field_border_detection.rs:116]: Done

File: crates/vision/src/line_detection.rs (hunk 2 above)
Comment (knoellle) [reply crates/vision/src/line_detection.rs]: Undone.

**Review by okiwi6 — APPROVED** [review]
Nice, LGTM

**Pull request merged**